File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float GetPositiveFloat(void);

int main(void)
{
    float number = GetPositiveFloat();
    int cents = round(number*100);
    int coins = 0;
//    printf("There are %i cents.\n", cents);
    
    for (int i = 0; cents > 0; )
    {
        if (cents >= 25)
        {
            cents = (cents - 25);
            coins = (coins + 1);
        }
        else if (cents >= 10)
        {
            cents = (cents - 10);
            coins = (coins + 1);
        }
        else if (cents >= 5)
        {
            cents = (cents - 5);
            coins = (coins + 1);
        }
        else
        {
            cents = (cents - 1);
            coins = (coins + 1);
        }
        i = i + 1;
    }
    
    printf("%i\n", coins);
}

float GetPositiveFloat(void)
{
    float n;
    do
    {
        printf("Please give me a positive dollar amount: ");
        n = GetFloat();
    }
    while (n < 0);
    return n;
}

File: pset6/greedy.py
import cs50

while True:
    print("Please give a positive cash amount (eg. 2.50): ", end="")
    n = cs50.get_float()
    if n < 0:
        print("Positive amounts only.")
    else:
        break

cents = round(n*100)
quarters = int(cents/25)
change = cents % 25

dimes = int(change/10)
change = change % 10

nickels = int(change/5)
pennies = change % 5

coins = quarters + dimes + nickels + pennies

print(coins)